bin/libs/fs.sh

```diff
@@ -32,19 +32,27 @@ $(echo "${children}")
 EOF
 }
 
-# Takes in the file that should be expanded and echos out the result, which the caller needs to read
 convert_to_absolute_path() {
-  file="${1}"
-
-  # If the value starts with a ~ for the home variable then evaluate it
-  file=$(echo "${file}")
-
-  # If the path is relative, then expand it
-  if [[ "${file}" != /* ]]; then
-    file="$(pwd)/${file}"
-  fi
-
-  echo "${file}"
+    # generate absolute path from relative path
+    # $1     : relative filename
+    # return : absolute path
+    orgPath="${1}"
+    if [ -d "$orgPath" ]; then
+        if [[ $orgPath = /* ]]; then
+            echo "$orgPath"
+        else
+            (cd "$orgPath"; pwd)
+        fi
+    elif  [ -f "$orgPath" ]; then
+        # file
+        if [[ $orgPath = /* ]]; then
+            echo "$orgPath"
+        elif [[ $orgPath == */* ]]; then
+            echo "$(cd "${orgPath%/*}"; pwd)/${orgPath##*/}"
+        else
+            echo "$(pwd)/$orgPath"
+        fi
+    fi
```

Comment:
Could we add a else statement here. If the input is not file and not dir, we fall back just echo out orgPath since we don’t know what to do.

Question, did you test it with symbolic link? the symlink could be a directory or a file. or if the relative path is under a symlink. for example `../a/abcd.pax`, which `../a` is a symlink of somewhere else. or `../a` is real, but `abcd.pax` is a symlink.

Reply:
Done
Assuming the pwd as `/var/zowe/vers/v2/bin`
`/var/zowe/vers/v2/bin/linkZowe.yaml` gives `/var/zowe/vers/v2/bin/linkZowe.yaml`
`linkZowe.yaml `gives` /SYSTEM/var/zowe/vers/v2/bin/linkZowe.yaml`
` ../bin/linkV2/zowe.yaml `gives `/SYSTEM/var/zowe/vers/v2/zowe.yaml`
`/var/zowe/vers/v2/bin/linkV2/bin/linkZowe.yaml` gives `/var/zowe/vers/v2/bin/linkV2/bin/linkZowe.yaml`
`../bin/linkV2/bin/linkZowe.yaml` gives `/SYSTEM/var/zowe/vers/v2/bin/linkZowe.yaml`